### DISCUSSION TITLE
Add device classes to `sensor` for weight

### DISCUSSION THREAD

**Discussion by nareddyt** [home-assistant/architecture#498] [Entity Models] [answered]
## Context
Smart bathroom scales, load cells, etc. read dimensions that HA can handle only as generic. I would like to introduce a device class for weight-based sensors.

## Proposal
Add 2 new device classes for weight-related items: `force` and `mass`

The original goal was to add a device class for weight. However, I found that the term `weight` is often misused. Weight is a `force`, but most people are really referring to `mass` when they measure their weight.

After [further reading](https://en.wikipedia.org/wiki/Mass_versus_weight), the SI units clearly define the distinction:
- `mass` (kg, lbs for imperial conversion): This is what people commonly refer to as weight. Can be used for measurements by smart bathroom scales, bed occupancy sensors, etc.
- `force` (N): Can be used by any sensors that are not manufactured to take into account gravity. Anything that senses force horizontally.

So I added support for both. Users can choose between the two to refer to the correct scientific term when they want to measure `weight`. 

PRs:
- Core PR: https://github.com/home-assistant/core/pull/45508
- home-assistant.io PR: https://github.com/home-assistant/home-assistant.io/pull/16301
- developers PR: https://github.com/home-assistant/developers.home-assistant/pull/779

## Consequences
Sensors can use more specific device class for "weight" measurement.

**Comment by MartinHjelmare** [comment]
It's easier to accept changes to the entity model if we can see an overview of the current integrations and also possible devices/brands in the wild and how many would benefit from the added device classes. Can you make a table or summary of this?

**Comment by nareddyt** [comment]
Good point. Here is a summary of what I've found:

- The [ESPHome integration](https://www.home-assistant.io/integrations/esphome/) allows connecting load cells to an ESP to measure `mass` ([reference](https://esphome.io/components/sensor/hx711.html)). Other sensors can be connected to measure force and/or pressure ([example](https://learn.sparkfun.com/tutorials/force-sensitive-resistor-hookup-guide?_ga=2.144539189.1078773881.1611594214-1375515631.1608939209)). I plan to implement https://github.com/esphome/feature-requests/issues/553 so device classes can be set for non-binary sensors directly in the ESP config. Both device classes will be useful here.
- The [Fitbit integration](https://www.home-assistant.io/integrations/fitbit/) calls the Fitbit API and retrieves `body/weight`. It is exposed as a sensor in HA. The `mass` device class will help here.
- Same for the [Smartthings integration](https://www.home-assistant.io/integrations/smartthings/), it exposes the `bodyWeightMeasurement` sensor.
- Same for [Withings](https://www.home-assistant.io/integrations/withings/). It has sensors for `fat_mass_lbs`, `bone_mass_lbs`, `muscle_mass_lbs`. Notice they all say `mass` in the name.

**Comment by blakadder** [comment]
Tasmota supports load cells.

Bluetooth scale data gathered with https://github.com/lolouk44/xiaomi_mi_scale

**Comment by nareddyt** [comment]
Friendly ping! Are there any concerns with this new device class? Would appreciate if the PRs can be merged, all the work is already done

**Comment by balloob** [comment]
I see a case for `mass` but I don't see any case for `force`. Let's just stick with `mass` for now?

**Reply by frenck** [reply]
I agree with that, the use case for mass is far bigger.

**Comment by MartinHjelmare** [comment]
Ok, I think we have support for `"mass"` at this point. Please go ahead and implement that.

**Reply by nareddyt** [reply]
SG, I'll remove the `force` implementation from the PRs.

**Comment by nareddyt** [comment]
All 3 PRs are updated. Please take a look

**Reply by ThomDietrich** [reply]
What happened here? All looks good down to the PR, then it never got merged 🤔

**Comment by emontnemery** [comment, marked as answer]
## Implemented solution

Add a `sensor` `device_class` for `weight` to home assistant core and frontend (+ documentation).
A `weight` sensor represents a measurement of an object's mass.
While `mass` would have been technically correct, `weight` was chosen in the end to better fit everyday language.

To solve the problem with different applications using different units for mass, without the need to add device classes for each of those applications, the unit of a `weight` sensor shall be configurable by the user in the same way as for temperature and pressure sensors.

Specific device classes for common applications may of course be added if there is a need.

### Relevant PRs
- [x] Frontend: https://github.com/home-assistant/frontend/pull/13897
- [x] Developer docs: https://github.com/home-assistant/developers.home-assistant/pull/1476
- [x] User docs: https://github.com/home-assistant/home-assistant.io/pull/24248
